cmd/landscaper-agent/app/app.go

```diff
@@ -49,20 +50,26 @@ func (o *options) run(ctx context.Context) error {
 	}
 
 	o.log.Info("Starting the controllers")
-	go func() {
-		if err := o.HostMgr.Start(ctx); err != nil {
-			o.log.Error(err, "error while running manager")
-			os.Exit(1)
-		}
-	}()
-	o.log.Info("Waiting for host cluster cache to sync")
-	if !o.HostMgr.GetCache().WaitForCacheSync(ctx) {
-		return errors.New("unable to sync host cluster cache")
-	}
+	eg, ctx := errgroup.WithContext(ctx)
 
-	o.log.Info("Cache of host cluster successfully synced")
-	if err := o.LsMgr.Start(ctx); err != nil {
-		return fmt.Errorf("error while running manager: %w", err)
+	if o.LsMgr != o.HostMgr {
+		eg.Go(func() error {
+			if err := o.HostMgr.Start(ctx); err != nil {
+				return fmt.Errorf("error while running host manager: %w", err)
+			}
+			return nil
+		})
+		o.log.Info("Waiting for host cluster cache to sync")
+		if !o.HostMgr.GetCache().WaitForCacheSync(ctx) {
+			return errors.New("unable to sync host cluster cache")
+		}
+		o.log.Info("Cache of host cluster successfully synced")
 	}
-	return nil
+	eg.Go(func() error {
+		if err := o.LsMgr.Start(ctx); err != nil {
+			return fmt.Errorf("error while running landscaper manager: %w", err)
+		}
+		return nil
+	})
+	return eg.Wait()
```

Comment:
It is not clear for me what you want to achieve with this construct, perhaps lack of knowledge. According to the documentation  `eg.Wait()` waits until all function calls returned. Could it not happen that e.g. `o.HostMgr.Start` fails but `o.LsMgr.Start` is still running resulting in a system with only one of the two manager working.

Reply:
no this cannot happen because the context of `eg, ctx := errgroup.WithContext(ctx)` will get automatically canceled if one of the functions started with `eg.Go` return an error.

See the errgroups docs:
```
// WithContext returns a new Group and an associated Context derived from ctx.
//
// The derived Context is canceled the first time a function passed to Go
// returns a non-nil error or the first time Wait returns, whichever occurs
// first.
```